### ISSUE TITLE
About the training batch size used in the low resource experiments

### ISSUE THREAD

**Issue by wangpf3** [amazon-research/tanl#5]
Hi! 
Thanks a lot for sharing the code! I'm trying to reproduce the low-resource experiments on the CoNLL04 dataset. Could you please provide the size of a training batch (per GPU and the number of GPUs used)? Would also appreciate a full training config!

**Comment by giove91** [comment]
Hi, thanks for your interest in TANL! :)

We used 8 GPUs, as in all other experiments except multi-task (see Appendix A in our paper).

The following is the config we used, which is almost identical to the `conll04_final` config already provided in the repo:
```
[conll04_fewdata]
datasets = conll04
model_name_or_path = t5-base
num_train_epochs = 200
max_seq_length = 256
max_seq_length_eval = 512
train_split = train,dev
per_device_train_batch_size = 8
per_device_eval_batch_size = 16
do_train = True
do_eval = False
do_predict = True
episodes = 1-10
num_beams = 8
train_subset = 1
```

This trains on the whole train+dev set. To train on less data, we changed `train_subset` to the following values: 0.5, 0.25, 0.125, 0.625, 0.03125, 0.01563, 0.00781. At the same time, we changed `num_train_epochs` to the following values: 400, 800, 1600, 2000, 2000, 2000, 2000. This is described in Appendix B in our paper.

Technically, we used `train_split = train_dev` together with the `train_dev.json` file downloaded from [SpERT](https://github.com/lavis-nlp/spert/blob/master/scripts/fetch_datasets.sh), because this made it easier to generate the same random subsets for us and for SpERT (to obtain Figures 2 and 3 in our paper). However, this should make little difference with respect to `train_split = train,dev`.

Note that the episode number is also used as random seed to generate the training subset.

Hope this helps!
